src/patchelf.cc

```diff
@@ -1745,21 +1754,81 @@ void ElfFile<ElfFileParamNames>::addNeeded(const std::set<std::string> & libs)
     changed = true;
 }
 
+// https://stackoverflow.com/a/478960/143733
+std::string exec(const char* cmd) {
+    std::array<char, 128> buffer;
+    std::string result;
+    std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd, "r"), pclose);
+    if (!pipe) {
+        throw std::runtime_error("popen() failed!");
+    }
+    while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
+        result += buffer.data();
+    }
+    return result;
+}
+
 template<ElfFileParams>
-void ElfFile<ElfFileParamNames>::printNeededLibs() // const
+void ElfFile<ElfFileParamNames>::shrinkWrap(std::map<std::string, std::string> & neededLibsToReplace, std::set<std::string> & neededLibsToAdd)
+{
+    const std::string interpreter = getInterpreter();
+    const std::vector<std::string> needed = getNeededLibs();
+    std::stringstream ss;
+    ss  << interpreter << " --list " << this->fileName;
```

Comment:
Is this glibc specific or does this work with musl libc as well?

Reply:
It does work with musl.
```
❯ /lib/ld-musl-x86_64.so.1 --list a.out
	/lib/ld-musl-x86_64.so.1 (0x7fb52fc0c000)
	libc.so => /lib/ld-musl-x86_64.so.1 (0x7fb52fc0c000)
```